### ISSUE TITLE
`sage.graphs`: Eliminate use of `sage.plot.colors` when not plotting

### ISSUE THREAD

**Issue by mkoeppe** [sagemath/sage#35777]
Currently, at least `sage.graphs.graph_coloring` uses it outside of a plotting context, which makes packaging (#35095) awkward

@dcoudert

**Comment by mkoeppe** [comment]
Yes, I saw that. Would it be possible to defer this assignment somehow?

**Comment by dcoudert** [comment]
Removing calls to `rainbow` from `sage.graphs.graph_coloring` is not easy. It would require to get surrounding methods in  (at least) `graph.py` to assign colors.

I will check the feasibility of making methods in `sage.graphs.graph_coloring` return color classes only (or a number for counting) and to add a method in charge of turning the color classes to the desired format: color classes (do nothing), mapping vertex to color, mapping color to vertices. This way, calls to rainbow would be contained in a single method.
